### ISSUE TITLE
Add autocompletion for nushell

### ISSUE THREAD

**Issue by TechWatching** [Azure/azure-cli#28652]
**Related command**
All commands

**Describe the solution you'd like**
To have the same auto complete feature for nushell than there is already for other shell (bash, powershell)

**Describe alternatives you've considered**
Implement myself a custom script to have auto complete, but it would never be up to date.

**Additional context**
I am using nushell  👉https://www.nushell.sh/

**Comment by bebound** [comment]
I'm not familiar with nushell, It seems that it can use fish completer.
`az` is supported by fish: 
https://github.com/fish-shell/fish-shell/blob/master/share/completions/az.fish
https://github.com/fish-shell/fish-shell/blob/master/share/functions/__fish_argcomplete_complete.fish

Ref: https://www.nushell.sh/cookbook/external_completers.html#fish-completer

**Comment by TechWatching** [comment]
Yes you are right but that's not optimal as it requires to have another dependency: fish installed. I'm working on Windows so fish is not an option for me
